src/map/lua/lua_baseentity.cpp

```diff
@@ -17709,6 +17710,98 @@ uint8 CLuaBaseEntity::getMannequinPose(uint16 itemID)
     return 0;
 }
 
+// Fishing Contest Utilities
+
+void CLuaBaseEntity::submitContestFish(uint32 score)
+{
+    if (auto* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))
+    {
+        fishingcontest::SubmitFish(PChar, score);
+    }
+}
+
+void CLuaBaseEntity::withdrawContestFish()
+{
+    if (auto* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))
+    {
+        fishingcontest::WithdrawFish(PChar);
+    }
+}
+
+uint32 CLuaBaseEntity::getContestScore()
+{
+    if (auto* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))
+    {
+        if (FishingContestEntry* entry = fishingcontest::GetPlayerEntry(PChar))
+        {
+            return entry->score;
+        }
+    }
+
+    return 0;
+}
+
+uint8 CLuaBaseEntity::getContestRank()
+{
+    if (auto* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))
+    {
+        if (FishingContestEntry* entry = fishingcontest::GetPlayerEntry(PChar))
+        {
+            return entry->contestRank;
+        }
+    }
+
+    return 0;
+}
+
+auto CLuaBaseEntity::getContestRewardStatus() -> sol::table
+{
+    auto reward = lua.create_table();
+
+    if (m_PBaseEntity->objtype == TYPE_PC)
+    {
+        auto*       PChar = static_cast<CCharEntity*>(m_PBaseEntity);
+        const char* Query = "SELECT contestrank, share "
+                            "FROM   fishing_contest_entries "
+                            "WHERE  name = '%s' "
+                            "AND    claimed != 1; ";
+
+        int32 ret = _sql->Query(Query, PChar->getName());
+
+        if (ret != SQL_ERROR && _sql->NumRows() > 0 && _sql->NextRow() == SQL_SUCCESS)
+        {
+            reward["rank"]  = _sql->GetUIntData(0);
+            reward["share"] = _sql->GetUIntData(1);
+        }
+    }
+
+    return reward;
+}
+
+auto CLuaBaseEntity::getContestRankHistory() -> sol::table
+{
+    auto table = lua.create_table();
+
+    if (m_PBaseEntity->objtype == TYPE_PC)
+    {
+        auto history = fishingcontest::GetFishingContestHistory(static_cast<CCharEntity*>(m_PBaseEntity));
+        table[1]     = get<0>(history);
+        table[2]     = get<1>(history);
+        table[3]     = get<2>(history);
+        table[4]     = get<3>(history);
+    }
+
+    return table;
+}
+
+void CLuaBaseEntity::claimContestReward()
+{
+    if (auto* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))
+    {
+        return fishingcontest::ClaimContestReward(PChar);
```

Comment:
function is marked as returning void, but there is a return here - which is intended?

Reply:
Ahhh whoops.  It should be void, which is why it worked and I didn't pick it up.  Will adjust.